spi.c

```diff
@@ -2,74 +2,76 @@
 #include "common.h"
 
 
-#define SPI_WAIT_FOR_AR(tries) { tries = 10; while (tries>0) { if (readl(marian->iobase + 0x70)==0x80000000) break; msleep(1); tries--; }}
-
-
-int marian_spi_transfer(struct marian_card* marian, uint16_t cs, uint16_t bits_write, uint8_t* data_write, uint16_t bits_read, uint8_t* data_read) {
-	int tries=10;
-	uint32_t buf=0;
+#define SPI_WAIT_FOR_AR(tries) { \
+					tries = 10; \
+					while (tries>0) { \
```

Comment:
odd indentation still.  and also not quite correct coding style, should be:
`while (tries > 0) { \`